core/src/main/java/com/graphhopper/routing/weighting/custom/FindMinMax.java

```diff
@@ -23,10 +23,11 @@ public class FindMinMax {
     public static void checkLMConstraints(CustomModel baseModel, CustomModel queryModel, EncodedValueLookup lookup) {
         if (queryModel.isInternal())
             throw new IllegalArgumentException("CustomModel of query cannot be internal");
-        if (queryModel.hasDistanceInfluence() && queryModel.getDistanceInfluence() < baseModel.getDistanceInfluence())
+        double qmDI = queryModel.hasDistanceInfluence() ? queryModel.getDistanceInfluence() : 0;
+        double bmDI = baseModel.hasDistanceInfluence() ? baseModel.getDistanceInfluence() : 0;
```

Comment:
Here the `0` is also the server-side default? Shouldn't there really only be one place where it appears: The code where the yaml definition of the model is translated to a Java object?

Reply:
Yes, I thought after reading the config file the distance influence field of the server-side model should either be 0 (if nothing was specified) or whatever was specified in the server config, but never `null`. And when we merge a request custom model with the one on the server-side we either use the one from the server (in case the requesting distance influence is `null` ) or use the value from the request (in case it was not `null`). 

So basically I think that `baseModel.getDistanceInfluence() == null` would be a programming error and can be treated as such just let the NullPointerException be thrown.

And for `qmDI` we don't need a default value, because when nothing is given we will use the server-side value. Written as code:

```java
        if (queryModel.getDistanceInfluence() != null && queryModel.getDistanceInfluence() < baseModel.getDistanceInfluence())
            throw new IllegalArgumentException("CustomModel in query can only use " +
                    "distance_influence bigger or equal to " + baseModel.getDistanceInfluence() + ", but was: " + queryModel.getDistanceInfluence());
```

Reply:
Ok, I see this is not the way we do it currently. We do not apply the default server-side value during the deserialization of `config.yml`, but only much later in `createWeightingParameters`. This way it is also applied when we use the Java API. And because we call FindMinMax even before that we need to pay attention that there is no  value yet :( Maybe it would be cleaner to use two separate classes for the server-side/profile CustomModel and the one used for the request.